src/Idris/Desugar.idr

```diff
@@ -82,72 +97,80 @@ extendSyn newsyn
            , "New (" ++ unwords (map show $ saveMod newsyn) ++ "): "
               ++ show (modDocstrings newsyn)
            ]
-         -- Check there are no conflicting definitions
-         let oldprefix = map (Prefix,) <$> (prefixes syn)
-         let newprefix = map (Prefix,) <$> (prefixes newsyn)
-         let oldinfix  = infixes syn
-         let newinfix  = infixes newsyn
-
-         let check = mapMaybe (\ entry@(k, ((_, v), (_, v'))) => entry <$ guard (k /= "-" && (v /= v')))
-                   . StringMap.toList
-         let inter = concatMap {t = List} {m = List _} check
-                   [ sharedSupport oldprefix newprefix
-                   , sharedSupport oldprefix newinfix
-                   , sharedSupport oldinfix  newprefix
-                   , sharedSupport oldinfix  newinfix
-                   ]
-         unless (null inter) $ recordWarning $ GenericWarn emptyFC $ unlines
-           $ ("Conflicting fixity declarations:" ::)
-           $ flip (concatMap {m = List _}) inter
-           $ \ (k, ((fc1, fix1, prec1), (fc2, fix2, prec2))) => pure $ unwords
-           [ "\{k}:"
-           , "\{show fix1} \{show prec1} (at \{show fc1})", "and"
-           , "\{show fix2} \{show prec2} (at \{show fc2})"
-           ]
 
-         put Syn ({ infixes $= mergeLeft (infixes newsyn),
-                    prefixes $= mergeLeft (prefixes newsyn),
+         put Syn ({ infixes $= merge (infixes newsyn),
+                    prefixes $= merge (prefixes newsyn),
                     ifaces $= merge (ifaces newsyn),
                     modDocstrings $= mergeLeft (modDocstrings newsyn),
                     modDocexports $= mergeLeft (modDocexports newsyn),
                     defDocstrings $= merge (defDocstrings newsyn),
                     bracketholes $= ((bracketholes newsyn) ++) }
                   syn)
 
+-- Check if a namespace is pointing toward a fixity
+-- An namespace can end in an operator fixity if it's in a %hide
+-- directive in order to remove the fixity from the module in order
+-- to better control parsing in the file.
+isOperatorFixityNS : Name -> Bool
+isOperatorFixityNS (OP (Just fx) n) = True
+isOperatorFixityNS (NS _ n) = isOperatorFixityNS n
+isOperatorFixityNS _ = False
+
 mkPrec : Fixity -> Nat -> OpPrec
 mkPrec InfixL p = AssocL p
 mkPrec InfixR p = AssocR p
 mkPrec Infix p = NonAssoc p
 mkPrec Prefix p = Prefix p
 
+
 toTokList : {auto s : Ref Syn SyntaxInfo} ->
+            {auto c : Ref Ctxt Defs} ->
             PTerm -> Core (List (Tok OpStr PTerm))
 toTokList (POp fc opFC opn l r)
     = do syn <- get Syn
          let op = nameRoot opn
-         case lookup op (infixes syn) of
-              Nothing =>
+         let foundFixities = lookupName (UN (Basic op)) (infixes syn)
+         let ignoreCompatible = nubBy (\(_, _, pr1), (_, _, pr2) =>
+                                    pr1 == pr2) foundFixities
+         case ignoreCompatible of
```

Comment:
This only compares infixes to infixes (and the later one only prefixes to prefixes)
whereas the previous approach compared all (in/pre)fixes and thus detected strictly
more conflicts.

Reply:
Will fix.

This also somewhat suggests that we can get rid of the `prefix`/`infix` maps in `SyntaxInfo` and handle the `-` special case as an intended conflict